### DISCUSSION TITLE
Upstram update

### DISCUSSION THREAD

**Discussion by srjlewis** [TruCopilot/couchbasev4-extension#2] [General]
ping @Geolim4 

Looks to be some movement from the couchbase devs at last
https://issues.couchbase.com/browse/CXXCBC-448

**Comment by Geolim4** [comment]
Hello, does this means that the fix proposal (if accepted) will allow us to not embed the "proc_fork" patch ?

**Reply by srjlewis** [reply]
In theory yes.

But in the real world I would like to refactor the driver so it only uses the patch on the effected versions only, this is so the driver will be compatable with all versions of couchbase4.

But I would wait and see what they come up with first, I am just glad to see movement on it.

**Comment by srjlewis** [comment]
@Geolim4

Just doing a quick update, couchbase have broke things again https://issues.couchbase.com/browse/PCBC-984 this time, they have put a blocking call into the php extension shutdown and locks up in the exit() call of a child process. 🤦

They a put this bug into there latest sprint, so hopefully fixed in the next version.

**Reply by Geolim4** [reply]
Why I'm not surprised ? :(
Thanks for the heads up !

**Reply by srjlewis** [reply]
Good news they will be doing a update in version 4.2.1 https://github.com/couchbase/couchbase-php-client/pull/157

They have introduced a new call to notifiy couchbase core of the fork.

I have also updated the driver and put it in a new branch for the time being https://github.com/PHPSocialNetwork/couchbasev4-extension/tree/post-ext-4-2-1

Please have nosey through.

**Reply by srjlewis** [reply]
@Geolim4 

They have relesed version 4.2.1, so I have tweeked the driver as needed.

All tests are pasing https://github.com/srjlewis/couchbasev4-extension/actions/runs/8831027270

When you are ready and happy, I will merge into master for release.

FYI : Couchbase have also put a document together to show all the changes from V3 to v4
https://github.com/couchbase/couchbase-php-client/blob/main/MIGRATION_FROM_3_x.md

**Comment by Geolim4** [comment]
Okay that's great, make the PR and I'll merge it tonight ;)

Thank you again @srjlewis :D

**Reply by srjlewis** [reply]
PR done

**Comment by srjlewis** [comment]
@Geolim4 a friendly nudge, if you could do the release tags would be graet 😁

**Comment by Geolim4** [comment]
Oh damn, remind me tonight, I completely forgot that 😔

**Reply by srjlewis** [reply]
Sorry I also kept forgetting in a evening to remind you, so a another freindly reminder 🤪

**Reply by Geolim4** [reply]
Can you put me a brief descriptive (for the changelog) of what you've changed since 9.2.0 tag please ?

Thanks :)

**Reply by srjlewis** [reply]
Yep sure, I did a breif update of the change log already, but it could do with a further update.

Introduced the ```Phpfastcache\Drivers\Couchbasev4\Driver::prepareToFork()```, due to the updated couchbase api for forking processes introduced in extension v4.2.1

I removed the ```(new CouchbaseConfig())->setDoForkDetection(true)```, as detection is now triggered by the above call

Updated the readme docs for the new api call

updated the testing to cover more situations and edge cases

Updated git actions to test on supported php versions, and on different extention versions.

I think that covers the bulk of it

**Reply by Geolim4** [reply]
Release done :)

https://github.com/PHPSocialNetwork/couchbasev4-extension/releases/tag/9.2.1

Sorry for late answer !

**Reply by srjlewis** [reply]
@Geolim4 thanks, dont worry yourself, we all have lifes outside of codeing, Thats why it has took me soon long to remind you for the release.
